### PR TITLE
[fix] Check global object, return empty location when undefined

### LINKED ISSUE #136: lolcation breaking in browser if loc param is falsy
https://github.com/unshiftio/url-parse/blob/714747f0163e7de088568000c892e64f04e49d9a/index.js#L53

[This line](https://github.com/unshiftio/url-parse/blob/714747f0163e7de088568000c892e64f04e49d9a/index.js#L53) breaks if the code is run in browser and if `loc` value which is passed to `lolcation` is falsy. It causes `Uncaught ReferenceError: global is not defined` error. I suggest finding a different way of checking if code is running in browser or under Node.

I suggest finding a different way of checking if code is running in browser or under Node.

Some more context: This happened to me when using Swagger-UI inside Angular 6 application. It was working fine on Angular CLI project before versions 6 because Angular CLI < 6 provided a shim for `global` which was removed in 6. Temporary fix is `window.global = window;`. Relevant issue in `angular-cli`: https://github.com/angular/angular-cli/issues/8160#issuecomment-386153833

### PR DESCRIPTION
**Pull request by ahonn**
fixed #136

### CODE REVIEW

**Comment by ahonn** [comment]
I cannot set the global variable to undefined. 
When I add tests and run `npm test`, it will failing. Because code coverage requires global objects.

```
/Users/ahonn/Desktop/url-parse/node_modules/nyc/index.js:422
  var coverage = global.__coverage__
                        ^

TypeError: Cannot read property '__coverage__' of undefined
    at coverageFinder (/Users/ahonn/Desktop/url-parse/node_modules/nyc/index.js:422:25)
    at NYC.writeCoverageFile (/Users/ahonn/Desktop/url-parse/node_modules/nyc/index.js:379:18)
    at EventEmitter.onExit.alwaysLast (/Users/ahonn/Desktop/url-parse/node_modules/nyc/index.js:360:11)
    at emitTwo (events.js:126:13)
    at EventEmitter.emit (events.js:214:7)
    at emit (/Users/ahonn/Desktop/url-parse/node_modules/nyc/node_modules/signal-exit/index.js:77:11)
    at processReallyExit [as reallyExit] (/Users/ahonn/Desktop/url-parse/node_modules/nyc/node_modules/signal-exit/index.js:138:3)
    at process.exit (internal/process.js:159:13)
    at process.on (/Users/ahonn/Desktop/url-parse/node_modules/mocha/bin/_mocha:40:13)
    at emitOne (events.js:116:13)
    at process.emit (events.js:211:7)
    at processEmit [as emit] (/Users/ahonn/Desktop/url-parse/node_modules/nyc/node_modules/signal-exit/index.js:149:35)
```

This change is work for me, it will not throw an error.
How should I add tests correctly?

**Comment by 3rd-Eden** [comment]
In Node.js, `global` will always be thing. So to check if the code will work without access to `global` you most likely need to create a new `vm` with a custom context. 
 https://nodejs.org/api/vm.html#vm_script_runincontext_contextifiedsandbox_options so you have control over which globals are available. 

So you might be able to pull this off by doing a `fs.readFileSync` of the `index.js` use that as source for the `vm.Script`, create a custom `vm. createContext` and run the script in that. Get the created URL instance from the sandbox and see if does what you expect it to do.

**Comment by ahonn** [comment]
I directly set global to undefined, restore it after the test.. Should I change to using `vm. createContext`?

**Comment by 3rd-Eden** [comment]
If it works, it works. So I'm fine with any solution.

**Comment by ahonn** [comment]
it works.

File: test/test.js [hunk 1]
```diff
@@ -29,6 +29,19 @@ describe('url-parse', function () {
     assume(url.hostname).to.be.a('string');
   });
 
+  it('parse when global variable is undefined', function () {
```
Comment (lpinca) [inline test/test.js]: Maybe change the title to something like "works when the `global` variable is not defined".

**Comment by 3rd-Eden** [comment]
Thanks for your contribution, it will end up in the next patch release.

**Pull request merged**